app/_how-tos/configure-oidc-with-refresh-token.md

```diff
@@ -0,0 +1,162 @@
+
+related_resources:
+  - text: Authentication in {{site.base_gateway}}
+    url: /gateway/authentication/
+  - text: OpenID Connect authentication flows and grants
+    url: /plugins/openid-connect/#authentication
+  - text: Refresh token grant
+    url: /plugins/openid-connect/#refresh-token-grant-workflow
+
+plugins:
+  - openid-connect
+
+entities:
+  - route
+  - service
+
+products:
+  - gateway
+
+works_on:
+  - on-prem
+  - konnect
+
+min_version:
+  gateway: '3.4'
+
+tools:
+  - deck
+
+prereqs:
+  entities:
+    services:
+      - example-service
+    routes:
+      - example-route
+  inline:
+    - title: Set up Keycloak
+      include_content: prereqs/auth/oidc/keycloak-password
+      icon_url: /assets/icons/keycloak.svg
+
+tags:
+  - authentication
+  - openid-connect
+
+tldr:
+  q: How do I use a refresh token to authenticate directly with my identity provider?
+  a: Using the OpenID Connect plugin, retrieve the refresh token and use it to authenticate with an identity provider (IdP) by passing the refresh token in a `Refresh-Token` header.
+
+cleanup:
+  inline:
+    - title: Clean up Konnect environment
+      include_content: cleanup/platform/konnect
+      icon_url: /assets/icons/gateway.svg
+    - title: Destroy the {{site.base_gateway}} container
+      include_content: cleanup/products/gateway
+      icon_url: /assets/icons/gateway.svg
+
+---
+
+## 1. Enable the OpenID Connect plugin with refresh tokens
+
+Using the Keycloak and {{site.base_gateway}} configuration from the [prerequisites](#prerequisites), 
+set up an instance of the OpenID Connect plugin with the refresh token grant.
+
+We're also enabling the password grant, as well as a refresh token header, so that you can test retrieving the token.
+
+Enable the OpenID Connect plugin on the `example-service` Service:
+
+{% entity_examples %}
+entities:
+  plugins:
+    - name: openid-connect
+      service: example-service
+      config:
+        issuer: ${issuer}
+        client_id:
+        - ${client-id}
+        client_secret:
+        - ${client-secret}
+        client_auth:
+        - client_secret_post
+        auth_methods:
+        - refresh_token
+        - password
+        refresh_token_param_type:
+        - header
+        refresh_token_param_name: refresh_token
+        upstream_refresh_token_header: refresh_token
+variables:
+  issuer:
+    value: $ISSUER
+  client-id:
+    value: $CLIENT_ID
+  client-secret:
+    value: $CLIENT_SECRET
+{% endentity_examples %}
+
+In this example:
+* `issuer`, `client ID`, `client secret`, and `client auth`: Settings that connect the plugin to your IdP (in this case, the sample Keycloak app).
+* `auth_methods`: Refresh token and password grant.
+* `refresh_token_param_type`: We want to search for the refresh token in headers only.
+
+## 2. Retrieve the refresh token
+
+Check that you can recover the refresh token:
+
+{% validation request-check %}
+url: /anything
+method: GET
+status_code: 200
+user: "john:doe"
```

Comment:
It took me a minute to understand that this was supposed to be the user and password set in Keycloak, I think we could use or environment variables or explicitly give the john:doe example in the prereq